src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.cpp

```diff
@@ -0,0 +1,116 @@
+tnsr::I<DataVector, Dim> auxiliary_field(
+    const tnsr::I<DataVector, Dim>& x,
+    const DataVector& precomputed_norm_square) noexcept;
+
+template <>
+tnsr::I<DataVector, 1> auxiliary_field<1>(
+    const tnsr::I<DataVector, 1>& x,
+    const DataVector& precomputed_norm_square) noexcept {
+  auto result = make_with_value<tnsr::I<DataVector, 1>>(x, 0.);
+  const auto& x_d = get<0>(x);
+  get<0>(result) = sqrt(precomputed_norm_square) *
+                   (0.25 - 3. * x_d + 7.5 * square(x_d) - 5. * pow(x_d, 3));
+  return result;
+}
+
+template <>
+tnsr::I<DataVector, 2> auxiliary_field<2>(
+    const tnsr::I<DataVector, 2>& x,
+    const DataVector& precomputed_norm_square) noexcept {
+  auto result = make_with_value<tnsr::I<DataVector, 2>>(x, 0.);
+  for (size_t d = 0; d < 2; d++) {
+    const auto& x_d = x.get(d);
+    const auto& x_p = x.get((d + 1) % 2);
+    result.get(d) =
+        sqrt(precomputed_norm_square) * x_p * (1. - x_p) *
+        (0.5 - 3.5 * x_d + 7.5 * square(x_d) - 5. * pow(x_d, 3) - x_p +
+         2. * x_d * x_p + square(x_p) - 2. * x_d * square(x_p));
+  }
+  return result;
+}
+
+}  // namespace
+
+template <size_t Dim>
+tuples::TaggedTuple<Field, AuxiliaryField<Dim>> Moustache<Dim>::field_variables(
+    const tnsr::I<DataVector, Dim>& x) const noexcept {
+  auto field = make_with_value<Scalar<DataVector>>(x, 1.);
+  for (size_t d = 0; d < Dim; d++) {
+    field.get() *= x.get(d) * (1. - x.get(d));
+  }
+  auto norm_square = make_with_value<DataVector>(get<0>(x), 0.);
+  for (size_t d = 0; d < Dim; d++) {
+    norm_square += square(x.get(d) - 0.5);
+  }
+  field.get() *= pow(norm_square, 3. / 2.);
+  return {std::move(field), auxiliary_field(x, norm_square)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>, ::Tags::Source<AuxiliaryField<1>>>
+Moustache<1>::source_variables(const tnsr::I<DataVector, 1>& x) const noexcept {
+  auto field_source = make_with_value<Scalar<DataVector>>(x, 1.);
+  const auto& x1 = get<0>(x);
+  field_source.get() *= (0.875 - 8.5 * x1 + 28.5 * square(x1) -
+                         40. * pow(x1, 3) + 20. * pow(x1, 4)) /
+                        abs(x1 - 0.5);
+  auto auxiliary_source =
+      make_with_value<tnsr::I<DataVector, 1, Frame::Inertial>>(x, 0.);
+  return {std::move(field_source), std::move(auxiliary_source)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>, ::Tags::Source<AuxiliaryField<2>>>
+Moustache<2>::source_variables(const tnsr::I<DataVector, 2>& x) const noexcept {
+  auto source = make_with_value<Scalar<DataVector>>(x, 1.);
+  const auto& x1 = get<0>(x);
+  const auto& x2 = get<1>(x);
+  const auto x2_sq = square(x2);
+  source.get() *= (6. * pow(x1, 5) - 2. * pow(x1, 6) +
+                   pow(x1, 4) * (-11. + 39. * x2 - 39. * x2_sq) +
+                   pow(x1, 3) * (12. - 78. * x2 + 78. * x2_sq) +
+                   pow(x1, 2) * (-7. + 65.5 * x2 - 104.5 * x2_sq +
+                                 78. * pow(x2, 3) - 39. * pow(x2, 4)) +
```

Comment:
Since you already have `x2_sq` why not do `x2_sq * x2` and `pow<2>(x2_sq)`?

Note: When you know the exponent at compile time you should prefer the `pow<#>` since it's more efficient.